linker/shared/src/main/scala/org/scalajs/linker/analyzer/InfoLoader.scala

```diff
@@ -116,13 +116,19 @@ private[analyzer] object InfoLoader {
 
     private def generateInfos(classDef: ClassDef): Infos.ClassInfo =  {
       val referencedFieldClasses = Infos.genReferencedFieldClasses(classDef.fields)
-      val methods = classDef.methods.map(methodsInfoCaches.getInfo(_))
 
-      val jsMethodProps = {
-        classDef.jsConstructor.map(jsConstructorInfoCache.getInfo(_)).toList :::
-        exportedMembersInfoCaches.getInfos(classDef.jsMethodProps)
+      prevMethodInfos = genMethodInfos(classDef.methods)
```

Comment:
It's a bit confusing that `genMethodInfos` uses the previous state of `prevMethodInfos` right before we reassign it. It doesn't help that there are two such methods that each access state that happens to be reassigned after them, but not the other state. It's not a big deal but it is a big hard to wrap my head around why it's safe.

It seems `prevMethodInfos` is in fact the only piece of state used by `genMethodInfos`. Did you consider explicitly passing `prevMethodInfos` as an argument to `genMethodInfos` and put the latter in a scope where it cannot access any state?

Concretely, amend this line to
```suggestion
      prevMethodInfos = genMethodInfos(classDef.methods, prevMethodInfos)
```
and move `genMethodInfos` outside of `private class ClassInfoCache`.

The same applies to `genJSMethodPropDefInfos` and possibly `genJSConstructorInfo`.

Reply:
yes, I agree. I didn't like this as well when I was writing it. Just didn't really follow through on the feeling.

I changed it.